### ISSUE TITLE
[macOS]Mac App failed to publish with error:"Unsupported Architectures. Your executable contained the following disallowed architectures: '[i386 Payload/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib)]"

### ISSUE THREAD

**Issue by GouriKumari** [xamarin/xamarin-macios#3367]
### Steps to Reproduce

1. Open  https://github.com/olegoid/SubmissionSamples/tree/master/MacCoolApp
2. Build and archive  
3. Upload macapp.pkg to iTunesConnect

### Expected Behavior
Upload should be successful

### Actual Behavior
Failed with an error "ERROR: ERROR ITMS-90240: "Unsupported Architectures. Your executable contained the following disallowed architectures: '[i386 (in com.xamarin360.macpublishworkflow.pkg/Payload/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib)]'. New apps submitted to the Mac App Store must support 64-bit starting January 2018, and Mac app updates and existing apps must support 64-bit starting June 2018."

### Environment
This issue occurs with both d15-5 stable and d15-6. 
https://gist.github.com/GouriKumari/cb70a4730ba9071697a2178ba7d78bf0

### Build Logs
Error Log: https://gist.github.com/GouriKumari/34cfbb7ca1341c2eb21f21f5f26dceae
build Log: https://gist.github.com/GouriKumari/f15bb9c9709ca31c24afb19fb0ef5fa8

### Example Project (If Possible)
https://github.com/olegoid/SubmissionSamples/tree/master/MacCoolApp

**Comment by dalexsoto** [comment]
mmm, I think this is just a release configuration issue, supported archs option here must be set to x86_64

<img width="958" alt="supportedarchs" src="https://user-images.githubusercontent.com/204671/35637352-8fd674c0-0679-11e8-9c67-e7b9947a546e.png">

**Comment by dalexsoto** [comment]
mmm never mind, I double-checked the configuration and seems to be this way already

```xml
<PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectTypeGuids>{A3F8F2AB-B479-4A4A-A458-A89E7DC349F1};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
    <ProjectGuid>{BDECBE4B-F410-45E3-BF6C-B7726D8A4F9F}</ProjectGuid>
    <OutputType>Exe</OutputType>
    <RootNamespace>MacCoolApp</RootNamespace>
    <MonoMacResourcePrefix>Resources</MonoMacResourcePrefix>
    <AssemblyName>MacCoolApp</AssemblyName>
    <TargetFrameworkVersion>v2.0</TargetFrameworkVersion>
    <TargetFrameworkIdentifier>Xamarin.Mac</TargetFrameworkIdentifier>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug</OutputPath>
    <DefineConstants>DEBUG;</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <UseRefCounting>true</UseRefCounting>
    <UseSGen>true</UseSGen>
    <IncludeMonoRuntime>false</IncludeMonoRuntime>
    <EnablePackageSigning>false</EnablePackageSigning>
    <CodeSigningKey>Mac Developer: Xamarin QA (JP4JS5NR3R)</CodeSigningKey>
    <EnableCodeSigning>true</EnableCodeSigning>
    <CreatePackage>false</CreatePackage>
    <PackageSigningKey>Developer ID Installer</PackageSigningKey>
    <CodeSignEntitlements>Entitlements.plist</CodeSignEntitlements>
    <XamMacArch>x86_64</XamMacArch>
    <CodeSignProvision>Automatic</CodeSignProvision>
    <Profiling>false</Profiling>
    <HttpClientHandler>HttpClientHandler</HttpClientHandler>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <Optimize>true</Optimize>
    <OutputPath>bin\Release</OutputPath>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <LinkMode>SdkOnly</LinkMode>
    <UseRefCounting>true</UseRefCounting>
    <UseSGen>true</UseSGen>
    <IncludeMonoRuntime>true</IncludeMonoRuntime>
    <CreatePackage>true</CreatePackage>
    <CodeSigningKey>3rd Party Mac Developer Application: Peter Beaman (ZYWNCWEKU5)</CodeSigningKey>
    <EnableCodeSigning>true</EnableCodeSigning>
    <EnablePackageSigning>true</EnablePackageSigning>
    <PackageSigningKey>3rd Party Mac Developer Installer: Peter Beaman (ZYWNCWEKU5)</PackageSigningKey>
    <XamMacArch>x86_64</XamMacArch>
    <CodeSignProvision>cc09f1f3-7939-4e0e-a5fc-807db3c1ef1c</CodeSignProvision>
    <CodeSignEntitlements>Entitlements.plist</CodeSignEntitlements>
    <Profiling>false</Profiling>
    <HttpClientHandler>HttpClientHandler</HttpClientHandler>
    <CodeSignResourceRules></CodeSignResourceRules>
  </PropertyGroup>
```

**Comment by GouriKumari** [comment]
I did check that supported architecture is already set to x86_64, https://github.com/olegoid/SubmissionSamples/blob/master/MacCoolApp/MacCoolApp/MacCoolApp.csproj#L53

**Comment by dalexsoto** [comment]
I cannot seem to repro with current master let me test with d-15

```
SubmissionSamples git:(master) file /Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MacOS/MacCoolApp
/Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MacOS/MacCoolApp: Mach-O 64-bit executable x86_64

SubmissionSamples git:(master) ✗ lipo -info /Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MacOS/MacCoolApp
Non-fat file: /Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MacOS/MacCoolApp is architecture: x86_64
```

**Comment by dalexsoto** [comment]
@GouriKumari if you have the binary handy could you run this command on the executable?

`lipo -info Release/MacCoolApp.app/Contents/MacOS/MacCoolApp`

**Comment by dalexsoto** [comment]
Ahh it seems it is not the main executable, sorry I should have reviewed the error message more carefully, the culprit is `libMonoPosixHelper.dylib`

```
file /Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib: Mach-O universal binary with 2 architectures: [i386:Mach-O dynamically linked shared library i386] [x86_64:Mach-O 64-bit dynamically linked shared library x86_64]
/Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib (for architecture i386):	Mach-O dynamically linked shared library i386
/Users/alex/Projects/Git/SubmissionSamples/MacCoolApp/MacCoolApp/bin/Release/MacCoolApp.app/Contents/MonoBundle/libMonoPosixHelper.dylib (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
```

**Comment by dalexsoto** [comment]
It seems we need to remove any other 32 bits from helper libs too, not just the main executable.

/cc @chamons

**Comment by chamons** [comment]
We might have to strip libMonoPosixHelper.dylib in release configurations...

**Comment by GouriKumari** [comment]
Verified submission with a nativeframework that has 32 bit and 64 bit part and submission wasn't allowed.
##Logs:
Build Log: https://gist.github.com/GouriKumari/f49cd6936f6f7a93bc6d020122ca1c24
Submission Log: https://gist.github.com/GouriKumari/5b8fdb69fcd8bbf29354948cdfefba2e

**Comment by spouliot** [comment]
The fix also need\ to cover the case of other architectures, e.g. https://bugzilla.xamarin.com/show_bug.cgi?id=42464

**Comment by chamons** [comment]
Fixed in master and PR'ed for 15.7:

https://github.com/xamarin/xamarin-macios/pull/3633
